commerce_option.services.yml

```diff
@@ -4,7 +4,7 @@ services:
     arguments: ['@entity_field.manager', '@entity_type.bundle.info', '@entity_type.manager', '@cache.data']
   commerce_option.option_resolver:
     class: Drupal\commerce_option\Resolver\OptionResolver
-    arguments: ['@entity_type.manager']
+    arguments: ['@entity_type.manager', '@commerce_option.option_field_manager']
```

Comment:
This is excellent that you're using the field manager instead to grab the option values. Great understanding of the code!

Reply:
Yep, We did not had to query db for getting the options info.